internal/service/alertconfiguration/data_source_alert_configuration.go

```diff
@@ -319,7 +319,7 @@ func outputAlertConfigurationResourceHcl(label string, alert *admin.GroupAlertsC
 		resource.SetAttributeValue("enabled", cty.BoolVal(*alert.Enabled))
 	}
 
-	for _, matcher := range alert.Matchers {
+	for _, matcher := range conversion.SlicePtrToSlice(alert.Matchers) {
```

Comment:
why this function over
```suggestion
	for _, matcher := range *alert.Matchers {
```

Reply:
```suggestion
	for _, matcher := range alert.GetMatchers() {
```
This will avoid a potential nil pointer dereference if `matchers` property is not defined.

Reply:
correct @AgustinBettati , that will be in the next PRs specific to each resource. 

@gssbzn it's what Agustin is saying, alert.Matchers could be nil so * panics.

with this PR we're trying to keep the same behavior as before the PR

Reply:
I'm with @AgustinBettati here that use the Getters, they are nil safe

Reply:
I'll use them in the next PR

Reply:
Why not avoid introducing `NonEmptySliceToSlicePtr`  in the first place? I'm still nitpicking how the test is written and a lot of those conversations disappear by not introducing the helper?

Reply:
at then I'm made the changes in this PR to use Get functions so no need to have SlicePtrToSlice method.  
thank you both!

Reply:
@gssbzn SlicePtrToSlice is already deleted, do you mean that or NonEmptySliceToSlicePtr?

Reply:
all good, my comment was before the deletion